lib/validators/chat_allow_uploads_validator.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class ChatAllowUploadsValidator
+  def initialize(opts = {})
+    @opts = opts
+  end
+
+  def valid_value?(value)
+    return true if value == "f"
+    if value == "t" && SiteSetting.secure_media && !ENV["DISCOURSE_ALLOW_UNSECURE_CHAT_UPLOADS"]
+      return false
+    end
+    true
+  end
+
+  def error_message
+    if SiteSetting.secure_media && !ENV["DISCOURSE_ALLOW_UNSECURE_CHAT_UPLOADS"]
```

Comment:
I think we should extract the conditional logic here in to a method instead of duplicating it in two different methods.